### ISSUE TITLE
Pressable elements not working in native-stack on Android devices with new architecture

### ISSUE THREAD

**Issue by DrZoidberg09** [software-mansion/react-native-screens#2219]
### Description

When using Android devices (iOS and Android emulator is fine) you cannot use onPress in the header bar. onPressIn and onPress out does work. This was tested mostly with Samsung devices.
This only happens with native-stack / Expo Router.

Please have a look here: 
https://github.com/expo/expo/issues/30032
https://github.com/react-navigation/react-navigation/issues/12039
https://github.com/facebook/react-native/issues/44643

### Steps to reproduce

1. Use react navigation native-stack with new arch enabled on RN 0.74
2. Go to a stack screen and try to press elements with onPress
3. onPress elements wont work. onPressIn and onPressOut does

### Snack or a link to a repository

https://github.com/DrZoidberg09/RN-Android-Touch-Issue/

For comparison a repo with bare RN, one with Expo and react navigation stack and one with Expo router / native-stack

### Screens version

3.32.0

### React Native version

0.74

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

None

### Architecture

Fabric (New Architecture)

### Build type

None

### Device

None

### Device model

Samsung S23, Samsung S8+

### Acknowledgements

Yes

**Comment by kkafar** [comment]
🫨

**Comment by kkafar** [comment]
Is that the thing only on real devices?

**Comment by DrZoidberg09** [comment]
Yes. So far, only tested with Samsung devices, incl. S24, S8+ and a Samsung tablet (dont remember the name)

**Comment by DrZoidberg09** [comment]
Plus everything pressable goes completely crazy in the screen (not only the header bar), if it is an actually build apk instead of an expo development build. However, not sure if this related.

**Comment by AnzeBlaBla** [comment]
Also happening here, can confirm we've only seen it happen on-device, specifically the S24.

**Comment by martinezguillaume** [comment]
I have the same issue here but with iOS (not tested on Android)
It is only present in production build
It seems that when `Pressable` is pressed, it is disappearing for 1 frame and reappear right after causing important flicker.
It is not happening all the time, for me, it is happening when I push a new screen and the previous screens has elements.

Screen <- Pressable working

EmptyScreen 
PushScreen <- Pressable is working

ScreenWithStuff
PushScreen <- Pressable is crazy

I hope it's understandable

Here is the behavior that I have
https://github.com/software-mansion/react-native-screens/assets/17292331/ff38698f-5a59-480e-bfeb-951e1f03b46f

Note : It is only happening in production build with new arch enabled

**Comment by tboba** [comment]
Hi @DrZoidberg09! Thanks for creating this issue. I can observe there are some several issues, related to this one:

https://github.com/software-mansion/react-native-screens/issues/2150

However, I didn't check if #2150 is related to the behavior of onPressIn/Out. Have you tried creating a navigation from stack navigator with `detachInactiveScreens` set to `false` inside the Navigator and checking if pressables also don't work there?

**Comment by DrZoidberg09** [comment]
@tboba I tried it with detachInactiveScreens true or false. It does not make any difference. Both does not work.

**Comment by tboba** [comment]
@DrZoidberg09 thanks for checking. This means that probably this issue is out of scope of react-native-screens (since the weird behavior still happens after disabling screens). I'll try to reproduce it on my side and re-check the scenario given above, but as I said, I would report this issue on react-navigation repository or official react-native one.

**Comment by DrZoidberg09** [comment]
I actually did open the issue at react-native first. They redirected me to react navigation, and they directed me here.

Feels a little bit like going from one doctor to another :)

For my understanding: Why this is not coming from screens? It can defitely be tracked down to the native-stack from react native navigation. Hence, they mentioned it is related to screens.

**Comment by tboba** [comment]
I understand your point of view 😄 That's why I asked if this issue also happens on the classic Stack Navigator with `detachInactiveScreens` turned off (which means that you're resigning from using react-native-screens). If this issue happens on views without screens, then this problem must be related to react-navigation/react-native, but that's of course worth checking.

**Comment by DrZoidberg09** [comment]
Oh, I noticed I misread your posting from above. I tried detachInactiveScreens true or false on native-stack navigator. Normal stack does work completely fine.
And the issue is only in the stack. The base view of the Tab for example is totally fine. Only if you navigate to a native-stack screen, the issues happen.

**Comment by DavidAmyot** [comment]
+1

Have the same issue since upgraded to latest Expo SDK 51 using **expo-router**. `onPress` buttons in the header do not work, but `onPressIn` does work. 

Works fine on the simulator, but on our Samsung _Galaxy Tab Active4 Pro 5g_ tablets, the issue is present (even with a development build instead of sdk).

One thing I saw too is that if I try to long press the button, the button's focused color does not stay active like it should, it flashes a fraction of a second.

**Comment by hyochan** [comment]
This is a real drawback of enabling newArch in Expo.

**Comment by alduzy** [comment]
Just like in https://github.com/software-mansion/react-native-screens/issues/1975#issuecomment-1819084348 Using `RectButton` from `react-native-gesture-handler` is a workaround in this case.

**Comment by hyochan** [comment]
> Just like in [#1975 (comment)](https://github.com/software-mansion/react-native-screens/issues/1975#issuecomment-1819084348) Using `RectButton` from `react-native-gesture-handler` is a workaround in this case.

Oh this works! Thank you

**Comment by jpdery** [comment]
In my case I'm not even able to focus on TextInputs

**Comment by WKampel** [comment]
onPress also not working for me. But onPressIn does work. I'm on the latest beta version and Android. But it seems to only be a problem for me in certain situations (like in a scrollview that's inside a drawer content).

**Comment by WKampel** [comment]
RectButton does work. But I'm not sure it can sub touchable opacity.

**Comment by mym0404** [comment]
I wonder software-mansion/react-native-screens#2292 fixes this. But it is shipped on `4.x` only and not in stable release (3.x).

**Comment by kkafar** [comment]
Let me update you on the status of the issue: 

This should be fixed by #2292 for pressables **inside screens contents**. If that's not the case we kindly ask you for a issue reproduction, so that we can improve the situation.

If the pressable is rendered **inside a native header**, it might still not work. We have a person working on this.

**Comment by devYonz** [comment]
I just tested with the release from 3 days ago with #2292  did not fix the hardware device issue. The funny thing I observed is that if you connect your phone to android studio and use a pointer to click on buttons they work fine.

**Comment by thibaultcapelli** [comment]
I have the same issue with new architecture enabled, only tested on Android Pixel 7a

### Package versions ###

    "@react-navigation/native": "^7.0.0",
    "@react-navigation/native-stack": "^7.0.0",
    "react": "18.3.1",
    "react-native": "0.76.1",
    "react-native-screens": "^4.0.0"
    
I also noticed the issue occur only with the onPress props but onPressIn also works fine. Also if I import Header from `@react-navigation/elements` it is also working fine.

So the issue looks related to the components in the header props only but not with the header prop itself

### Sample ###

```
import React from 'react';
import {Pressable, Text, View} from 'react-native';

import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {Header} from '@react-navigation/elements';

function HomeScreen() {
  return (
    <View>
      <Text>Home Screen</Text>
    </View>
  );
}

const MyStack = createNativeStackNavigator({
  screens: {
    Home: HomeScreen,
  },
});

function App() {
  return (
    <NavigationContainer>
      <MyStack.Navigator
        screenOptions={{
          gestureEnabled: true,
          fullScreenGestureEnabled: true,
          headerLeft: () => (
            <Pressable
            // onPress={() => console.info('Hello2')} // => KO
            // onPressIn={() => console.info('Hello2')} // => OK
            >
              <Text>Hello2</Text>
            </Pressable>
          ),
        }}>
        <MyStack.Screen name={'HomeScreen'} component={HomeScreen} />
      </MyStack.Navigator>
    </NavigationContainer>
  );
}

function App() {
  return (
    <NavigationContainer>
      <MyStack.Navigator
        screenOptions={{
          // => OK
          header: () => (
            <Header
              title={'Title'}
              headerLeft={() => {
                return (
                  <Pressable onPress={() => console.info('Hello')}>
                    <Text>Hello</Text>
                  </Pressable>
                );
              }}
            />
          ),
        }}>
        <MyStack.Screen name={'HomeScreen'} component={HomeScreen} />
      </MyStack.Navigator>
    </NavigationContainer>
  );
}
```

Hope it helps

**Comment by zingzongzung** [comment]
Hi guys, 

I am new to react native world and while learning about the navigation details I faced into the same issue.

I have isolated both   "@react-navigation/drawer": "^7.0.3" and "@react-navigation/native-stack": "^7.0.2". With the native-stack one the header press are not working like this issue refers. 
With only the react-navigation/drawer the presses anywhere on the screen work as expected
However when I nested drawer navigation within  stack navigation, the same pressable text defined on HomeScreen wouldnt work.
Please refer to a sample simple project to see it in action.
https://github.com/zingzongzung/ReactNativeScreen

**Comment by Yurii-Lutsyk** [comment]
Any updates? I can't replace the Pressable component with the RectButton?

**Comment by zingzongzung** [comment]
@kkafar that will fix pressables on the header but will that fix the pressable in the body in case there is a drawer navigation nested with the stack navigation like I refered on my previous comment?

**Comment by MikeKovarik** [comment]
+1. I started migrating to RN 0.76 & new arch but this is blocking me. Pixel 4, real device.

**Comment by MikeKovarik** [comment]
> Hi any solution? the button or pressable press not work correctly in header.

Here's my workaround

```js
// workaround
const tap = Gesture.Tap().runOnJS(true).onEnd(onPress as any)

return (
    <GestureDetector gesture={tap}>
        <RNPressable {...props} />
    </GestureDetector>
)

// orignal impl. restore once RN new-arch bug is fixed
//return <RNPressable onPress={onPress} {...props} />
```

**Comment by Yurii-Lutsyk** [comment]
> ```js
> return (
>     <GestureDetector gesture={tap}>
>         <RNPressable {...props} />
>     </GestureDetector>
> )
> ```
This works for me, but just as a temporary solution until this issue is fixed

**Comment by acrabb** [comment]
+1 +1 +1 🙏 

I just ran into this issue upgrading to Expo 52 with New Architecture.
How is this not fixed yet?? 😅 
onPress in headers seems like a pretty common thing!

**Comment by beinoriusju** [comment]
Expanding on temp solution that works

```js
import { Pressable } from 'react-native';
import {Gesture, GestureDetector} from 'react-native-gesture-handler'

const RNPressable = ({ onPress, ...props }) => {
    const tap = Gesture.Tap().runOnJS(true).onEnd(onPress)

    return  <GestureDetector gesture={tap}>
    <Pressable {...props} />
</GestureDetector>
}

export default RNPressable;

**Comment by Tencryn** [comment]
I cannot make sense of this. It's weird that onPressOut will work with TouchableOpacity, but not with Button. So the Button component is not usable at all, while TouchableOpacity has a simple workaround by using onPressOut instead of onPress.

**Comment by smboy86** [comment]
> I cannot make sense of this. It's weird that onPressOut will work with TouchableOpacity, but not with Button. So the Button component is not usable at all, while TouchableOpacity has a simple workaround by using onPressOut instead of onPress.

its' same solution, 
in my case 
onPress => onPressOut 
clear..

from
`
<Button 
   ...
   onPress={...}
   ...
`

to
`
<Button 
   ...
   onPressOut={...}
   ...
`

**Comment by EdwardNavarro** [comment]
> > I cannot make sense of this. It's weird that onPressOut will work with TouchableOpacity, but not with Button. So the Button component is not usable at all, while TouchableOpacity has a simple workaround by using onPressOut instead of onPress.
> 
> its' same solution, in my case onPress => onPressOut clear..
> 
> from `<Button ... onPress={...} ...`
> 
> to `<Button ... onPressOut={...} ...`

That worked for me, thanks!

**Comment by ycldz** [comment]
import { TouchableOpacity } from 'react-native-gesture-handler';

this code does not work in Stack.Screen on iphone 11 -> ios 18.0.1. The best option seems to be onPressIn.

**Comment by kriit24** [comment]
same thing in ios [https://github.com/expo/expo/issues/32927](https://github.com/expo/expo/issues/32927)

**Comment by rankangkang** [comment]
same issue on ios with expo 52, looking forward to the fix so much!!!

**Comment by rankangkang** [comment]
While debugging, I noticed an issue related to `ScrollView`. Here's the scenario: assume the `ScrollView` contains a list of `TouchableOpacity` elements (long enough to make the page scrollable). When the `ScrollView` is in its initial state (not scrolled), the `onPress` events of the `TouchableOpacity` elements trigger as expected. However, after scrolling down a certain distance, the onPress events occasionally fail to trigger. Scrolling the `ScrollView` back to its initial position restores the normal functionality of the `onPress` events.

This led me to a hypothesis: the scrolling offset introduced by `ScrollView` might interfere with the touchable components' hitbox detection, causing the `onPress` events to fail intermittently.

For context, I tested this on an iPhone X running iOS `16.7.1`, and the issue seems more pronounced on older devices.

**Comment by nikodunk** [comment]
Thanks so much for the fix! I'm leaving testing feedback in the PR.

**Comment by cyburns** [comment]
I am having the same issue. switching the press event from `onPress` to `onPressIn` seems to have fixed the issue although it is not ideal.

**Comment by kriit24** [comment]
problem is not only that onPress does not work, but onPressIn also does not work all time. when u in development mode and usb cable is connected then 10+ times clicked onPressIn also stops working. i dont know, does this happens in live also, but at this moment I don't dare build anything to live.

**Comment by antoinerey** [comment]
Could it be possible that despite the fix for screen content, and the latest one for header content, this issue still impacts form sheets?

I tried different presentation modes (e.g. regular screens, modals, and form sheets), and it seems that pressable elements in form sheets are still kinda hard to press. Both regular screen and modal presentation modes are working perfectly, but not sheets.

**Comment by kkafar** [comment]
@antoinerey what platform do you speak of? Android or iOS?

**Comment by antoinerey** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/3336873?s=64&amp;u=2991705724a28c40e76142293f1f319537cd0b62&amp;v=4">@antoinerey](https://github.com/antoinerey?rgh-link-date=2025-01-17T17%3A25%3A12.000Z) what platform do you speak of? Android or iOS?

Android 🤗

**Comment by antoinerey** [comment]
Sorry for the late reproduction, but I've _not_ been able to actually reproduce on a minimal application. There might be something conflicting in our production application, but I've not had time to deeply investigate, so I'm not sure what might be the root cause.

Feel free to discard my report as it seems invalid. 🙇

**Comment by pknrs** [comment]
This is still an issue as of now

**Comment by kkafar** [comment]
@pknrs Can you provide a reproducer using 4.6.0 version of screens?

**Comment by Kartik4152** [comment]
Facing the same issue.

Observations till now:
- Clicking on it using a cursor works (using scrpy to stream app to mac)
- onPressOut gets called 200-250ms after onPressIn automatically even if I don't lift off my finger.
- onPress gets called if I spam the Pressable 20-30 times sometimes, sometimes never gets called at all

**Comment by kkafar** [comment]
In cases you're still experiencing issues, please provide a reproduction code, together with package versions you're using and preferably video of the bug occurring, if feasible. As of now I consider the bug fixed with release of 4.6.0.

**Comment by gersomvg** [comment]
I've been spending hours on this already :( No idea what is causing this.

**Comment by freeboub** [comment]
I can confirm the issue is still present. It is hard to provide on a sample as (on our project) it is reproduced only in release mode with a real device ...

**Comment by perttuvirkki** [comment]
Any update on this? Pretty much stuck with my app updates as this is breaking the buttons. I'm also having issues with Pressables inside ScrollView components.

**Comment by cyburns** [comment]
Still an issue for me as well

**Comment by kkafar** [comment]
I can only repeat my previous comment https://github.com/software-mansion/react-native-screens/issues/2219#issuecomment-2633587313

**Comment by mwerder** [comment]
can confirm it is fixed. Updating `react-native-screens` to `4.7.0` fixed it for me. Thanks

**Comment by cyburns** [comment]
> can confirm it is fixed. Updating `react-native-screens` to `4.7.0` fixed it for me. Thanks

I just upgraded from `4.4.0` to `4.7.0` and am having the same issue. In fact it seems to be an issue on IOS now as well.

**Comment by kkafar** [comment]
I can only repeat my previous comment https://github.com/software-mansion/react-native-screens/issues/2219#issuecomment-2633587313

**Comment by cyburns** [comment]
## Pressabbles not working within header

Sure, here is a minimum reproduction of where the issue is coming from. I'd also like to note that after upgrading from react-native-screen 4.4.0 to 4.7.0 the buttons changing widths is an issue. If the button gets wider it gets cut off.

Swapping onPress with onPressIn seemed to sometimes fix the issue in 4.4.0. After upgrading to 4.7.0 the issue seems to be on IOS (simulator) now as well where it was only on physical Android devices before. 

## Dependencies

```json
    "react": "18.3.1",
    "react-native": "0.76.5",
    "react-native-gesture-handler": "~2.20.2",
    "react-native-pager-view": "6.5.1",
    "react-native-reanimated": "~3.16.1",
    "react-native-screens": "^4.7.0",
    "react-native-tab-view": "^3.5.2",
```

## Minimum code example

```typescript
import { useNavigation } from 'expo-router';

const ParentComponent = () => {
  const [isEdit, setIsEdit] = useState(false);

  const nav = useNavigation();

  useLayoutEffect(() => {
    nav.setOptions({
      title: 'Device Details',
      headerLeft: () => (
        <CancelOrBackHeaderLeft isEdit={isEdit} handleEdit={handleEdit} />
      ),
      headerRight: () => (
        <EditHeaderRight
          isEdit={isEdit}
          hasEditPrivileges={hasEditPrivileges}
          isEditLoading={isEditLoading}
          handleEdit={handleEdit}
          handleSaveEdits={handleSaveEdits}
          disabled={
            !newDeviceName || !validateTags() || isEditLoading || loading
          }
        />
      ),
    });
  }, [
    nav,
    isEdit,
    isEditLoading,
    hasEditPrivileges,
    handleEdit,
    handleSaveEdits,
    newDeviceName,
    validateTags,
    loading,
  ]);

  return <></>;
};

const CancelOrBackHeaderLeft = ({
  isEdit,
  handleEdit,
}: DeviceDetailsHeaderProps) => {
  return (
    <View style={[a.flex_shrink]}>
      {!isEdit ? (
        <ScreenBack />
      ) : (
        <TouchableOpacity onPress={handleEdit} style={{ minWidth: 100 }}>
          <Text style={styles.headerCancel}>{t('general.cancel')}</Text>
        </TouchableOpacity>
      )}
    </View>
  );
};

const EditHeaderRight = ({
  isEdit,
  hasEditPrivileges,
  isEditLoading,
  handleEdit,
  handleSaveEdits,
  disabled,
}: DeviceDetailsHeaderProps) => {
  return (
    <View style={[a.flex_shrink]}>
      {hasEditPrivileges && (
        <Fragment>
          {!isEdit ? (
            <TouchableOpacity onPress={handleEdit}>
              <Ionicons
                name="create-outline"
                size={24}
                color={isEdit ? Colors.lightGray : Colors.black}
              />
            </TouchableOpacity>
          ) : (
            <TouchableOpacity
              style={{
                backgroundColor: disabled ? Colors.darkGray : Colors.black,
                borderRadius: 99,
                paddingVertical: 4,
                paddingHorizontal: 14,
                minWidth: 100,
              }}
              onPress={handleSaveEdits}
              disabled={disabled}
            >
              {isEditLoading ? (
                <LoadingDots color={Colors.white} />
              ) : (
                <Text style={[a.btnText, { color: Colors.white }]}>
                  {t('general.save')}
                </Text>
              )}
            </TouchableOpacity>
          )}
        </Fragment>
      )}
    </View>
  );
};

```

**Comment by kkafar** [comment]
Thanks for the repro! I'll get back to you once I investigate it

**Comment by yeliu84** [comment]
> Thanks for the repro! I'll get back to you once I investigate it

I'm facing the same issue as @cyburns, any updates?

**Comment by damiensedgwick** [comment]
All of the internal testers on android have reported header buttons not working at all despite being fine on emulators.

**Comment by yeliu84** [comment]
> All of the internal testers on android have reported header buttons not working at all despite being fine on emulators.
> 
> Upgrading to 4.7.0 for screens did not help me.

I'm having this issue on 4.9.1. I think this issue should be re-opened for further investigation.

**Comment by mikelgarciaurbina** [comment]
Same issue here. I tried to update to 4.9.1 but I am still with the issue. I see that if I set headerTransparent to false it solves all of my issues, except the custom back button on the header. Maybe is something related to the position absolute of the header?

**Comment by atgctg** [comment]
Interesting, disabling `headerTransparent` fixes things for me as well @mikelgarciaurbina.

Going to use `onPressIn` as a workaround for now.

**Comment by yeliu84** [comment]
> Same issue here. I tried to update to 4.9.1 but I am still with the issue. I see that if I set headerTransparent to false it solves all of my issues, except the custom back button on the header. Maybe is something related to the position absolute of the header?

Set `headerTransparent: false` also made `onPress` work for me. Hope this can help @kkafar 's investigation.

**Comment by hirbod** [comment]
For anybody stuck on this: 

Use Pressable, RectButton or BorderlessButton exported by react-native-gesture-handler for now - don't use onPressIn.
Here is my thread on it. https://x.com/hirbod_dev/status/1900204098344948015

It's a clean and feasible workaround to get unblocked (and the pressables are better anyway)

**Comment by yeliu84** [comment]
> For anybody stuck on this:
> 
> Use Pressable, RectButton or BorderlessButton exported by react-native-gesture-handler for now - don't use onPressIn. Here is my thread on it. https://x.com/hirbod_dev/status/1900204098344948015
> 
> It's a clean and feasible workaround to get unblocked (and the pressables are better anyway)

That's great workaround! I however still have one issue remaining is that the `<TextInput>` won't get focused when tapping on it with transparent header.

@kkafar this issue really needs some attention.

**Comment by DOWSCZ** [comment]
Guys, how can you develop without onPress working across the entire React Native and mobile view? I'm just curious—do you all work exclusively on the emulator? I tried setting headerTransparent to false, but it didn't help. On mobile, onPress works for 0.5 seconds after start and then stops working.

**Comment by jiroscripts** [comment]
@DOWSCZ 

I guess there is many scenarios. In our app we completely avoided to use react native modals in favor of react navigation modals. It also looks like the PR for pressable in headers fixed our problem despite the fact others persons reported others bugs. I also had to remove a "fetchMore" button in a scrollview by fetching items automatically on scroll end. For now we compose with these changes in our app to maintain a fragile balance. 

I'm following these issues for weeks and I will try to update my libs once there will be more positive feedbacks about all this. Honestly I feel sorry for the maintenairs cause new arch bugs looks like hell but this is a shame that these problems have been going for so long... I love so much React Native and the effort that Meta and community puts in the development of the new arch. But when I see that I cannot stop thinking that it will loose so many developers and I'm really sad about this cause I thinks RN should have a very bright future ahead of it.

**Comment by peckomingo** [comment]
I have the same issue. It does not work properly with `headerTransparent: true`. I believe it is related to the gesture recognition system because if you tap very carefully on a Pressable without moving your finger in any x or y, it works in most cases. However in reality, of the time, the finger is not the same coordinate and moves even by 3 pixels.

Therefore, every pan gesture prevents the successful trigger of the `onPress` event. Because it is interpreted or interrupted by another event type?

**Comment by fabrizioiacobucci** [comment]
Same issue here. Just added a simple button in the header and it's not firing the OnPress callback

**Comment by kubijo** [comment]
I've resorted to replacing all Pressable instances (and stuff related to it) with the replacements from the gesture handler `react-native-gesture-handler` … those seem to work fine. Only shortcoming compared to `react-native` version is that the pressable state does not have hover, but that did not concern me that much considering the usecase

**Comment by jiroscripts** [comment]
I also replaced it with the Pressable from gesture-handler and it works fine too.

**Comment by kkafar** [comment]
Hey, last week I've discovered by accident that the `gesture-handler` cancels the events from pressables, when they do have gesture detector above them in the view hierarchy. 

I've released a patch in 4.11.0-beta.0, therefore it might be worth to try it out. 

I haven't managed to reproduce "complete unresponsive" pressables though. This stays a mystery.

**Comment by elkinjosetm** [comment]
I wonder why this issue remains closed, when clearly is still happening

**Comment by Gopinatha-L** [comment]
#2855

**Comment by cjpete** [comment]
> Hey, last week I've discovered by accident that the `gesture-handler` cancels the events from RN pressables, when they do have gesture detector above them in the view hierarchy.
> 
> I've released a ~patch~ workaround in 4.11.0-beta.0, therefore it might be worth to try it out.
> 
> I haven't managed to reproduce "complete unresponsive" pressables though. This stays a mystery.

I've tried `4.11.0-beta.2` and it's still an issue for me. Both 4.10 and 4.11 occasionally capture a tap, but I can't figure out any kind of pattern.

**Comment by kkafar** [comment]
@cjpete are you able to prove me with reproduction?

**Comment by elkinjosetm** [comment]
It's certainly difficult to recreate, but what I've notice in our app is that it happens when you press a button inside an screen presented as modal. Also noticed that increasing font size at the OS level increases the failure rate, but still hit and miss

**Comment by haeniya** [comment]
> > Hey, last week I've discovered by accident that the `gesture-handler` cancels the events from RN pressables, when they do have gesture detector above them in the view hierarchy.
> > I've released a ~patch~ workaround in 4.11.0-beta.0, therefore it might be worth to try it out.
> > I haven't managed to reproduce "complete unresponsive" pressables though. This stays a mystery.
> 
> I've tried `4.11.0-beta.2` and it's still an issue for me. Both 4.10 and 4.11 occasionally capture a tap, but I can't figure out any kind of pattern.

I had the same issue @cjpete, sometimes `onPress` was firing and sometimes not, without any pattern.
In my case, the height of one of my custom components in the header was too large. Reducing the height resolved the issue.
Maybe that helps someone.

**Comment by kkafar** [comment]
@elkinjosetm what do you mean by "`screen` presented as modal" on Android? `presentation: 'modal'`?

**Comment by elkinjosetm** [comment]
@kkafar yes, when navigating to a screen presented as modal, anything inside that modal screen that is pressable, in release mode, behaves erratic, I've notice that it only happens in release mode, but still haven't figure out the reason, seems to only happens on physical devices, hope this helps.

Using buttons and pressable from react-native-gesture-handler seems to resolves the issue in some cases, but we shouldn't have to use a third-party library to create a simple button, but is a workaround for the moment.

**Comment by RahulMandyal1** [comment]
+1
Pressable is not working — not sure if this is a react-native-screens specific issue. It's already closed here:
[Pressable elements not working in native-stack on Android with new architecture](https://github.com/expo/expo/issues/33072)

**Comment by marijang** [comment]
+1 also not working onPress on samsung phones

**Comment by kkafar** [comment]
I need a reproduction guys. "It's not working" is not enough for us to work with, since we do not observe the problem on our end.

**Comment by marijang** [comment]
`{
  "name": "scipalote",
  "main": "expo-router/entry",
  "version": "1.0.0",
  "scripts": {
    "start": "expo start",
    "reset-project": "node ./scripts/reset-project.js",
    "prebuild-android": "expo prebuild --clean --platform android",
    "android-device": "expo run:android --device",
    "build-android": "cd android && ./gradlew assembleRelease",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "test": "jest --watchAll",
    "lint": "expo lint"
  },
  "jest": {
    "preset": "jest-expo"
  },
  "dependencies": {
    "@expo/react-native-action-sheet": "^4.1.1",
    "@expo/vector-icons": "^14.0.2",
    "@gorhom/bottom-sheet": "^5.1.4",
    "@hookform/resolvers": "^5.0.1",
    "@kolking/react-native-avatar": "^2.1.5",
    "@react-native-community/netinfo": "^11.4.1",
    "@react-navigation/bottom-tabs": "^7.3.13",
    "@react-navigation/native": "^7.1.9",
    "@rn-primitives/alert-dialog": "^1.1.0",
    "@rn-primitives/avatar": "^1.1.0",
    "@rn-primitives/dropdown-menu": "^1.1.0",
    "@rn-primitives/hooks": "^1.1.0",
    "@rn-primitives/label": "^1.1.0",
    "@rn-primitives/portal": "^1.2.0",
    "@rn-primitives/slot": "^1.1.0",
    "@rn-primitives/types": "^1.1.0",
    "@roninoss/icons": "^0.0.4",
    "@shopify/flash-list": "1.8.0",
    "@tanstack/react-query": "^5.76.1",
    "axios": "^1.9.0",
    "class-variance-authority": "^0.7.1",
    "clsx": "^2.1.1",
    "date-fns": "^4.1.0",
    "date-fns-tz": "^3.2.0",
    "eventemitter3": "^5.0.1",
    "expo": "~53.0.9",
    "expo-blur": "~14.1.4",
    "expo-camera": "~16.1.6",
    "expo-constants": "~17.1.6",
    "expo-dev-client": "~5.1.8",
    "expo-font": "~13.3.1",
    "expo-haptics": "~14.1.4",
    "expo-image": "~2.1.7",
    "expo-linking": "~7.1.5",
    "expo-navigation-bar": "~4.2.4",
    "expo-router": "~5.0.7",
    "expo-splash-screen": "~0.30.8",
    "expo-status-bar": "~2.2.3",
    "expo-symbols": "~0.4.4",
    "expo-system-ui": "~5.0.7",
    "expo-web-browser": "~14.1.6",
    "fast-xml-parser": "^5.2.3",
    "i18next": "^25.0.2",
    "lucide-react-native": "^0.503.0",
    "moti": "^0.30.0",
    "nativewind": "^4.1.23",
    "prettier-plugin-tailwindcss": "^0.6.11",
    "react": "19.0.0",
    "react-dom": "19.0.0",
    "react-hook-form": "^7.56.3",
    "react-i18next": "^15.2.0",
    "react-native": "0.79.2",
    "react-native-date-picker": "^5.0.12",
    "react-native-flash-message": "^0.4.2",
    "react-native-gesture-handler": "~2.25.0",
    "react-native-ios-context-menu": "^3.1.2",
    "react-native-ios-utilities": "^5.1.5",
    "react-native-keyboard-controller": "^1.17.1",
    "react-native-mmkv": "^3.2.0",
    "react-native-notifier": "^2.0.0",
    "react-native-page-indicator": "^2.4.2",
    "react-native-reanimated": "~3.17.5",
    "react-native-safe-area-context": "5.4.0",
    "react-native-screens": "~4.10.0",
    "react-native-svg": "15.12.0",
    "react-native-swiper-flatlist": "^3.2.5",
    "react-native-tab-view": "^4.1.0",
    "react-native-uitextview": "^2.0.3",
    "react-native-uuid": "^2.0.3",
    "react-native-web": "~0.20.0",
    "react-native-webview": "13.13.5",
    "scandit-datacapture-frameworks-core": "7.2.2",
    "scandit-react-native-datacapture-barcode": "7.2.2",
    "scandit-react-native-datacapture-core": "7.2.2",
    "scandit-react-native-datacapture-id": "7.2.2",
    "scandit-react-native-datacapture-parser": "7.2.2",
    "tailwind-merge": "^3.2.0",
    "tailwind-variants": "^1.0.0",
    "tailwindcss-animate": "^1.0.7",
    "xmlbuilder2": "^3.1.1",
    "zod": "^3.24.4",
    "zustand": "^5.0.4"
  },
  "devDependencies": {
    "@babel/core": "^7.26.9",
    "@dev-plugins/react-query": "~0.3.1",
    "@types/jest": "^29.5.14",
    "@types/react": "~19.0.10",
    "eslint": "^9.25.0",
    "eslint-config-expo": "~9.2.0",
    "jest": "^29.7.0",
    "jest-expo": "~53.0.5",
    "prettier-plugin-tailwindcss": "^0.6.11",
    "tailwindcss": "^3.4.17",
    "typescript": "^5.8.3"
  },
  "private": true
}
`

**Comment by AlexMachin1997** [comment]
> `{ "name": "scipalote", "main": "expo-router/entry", "version": "1.0.0", "scripts": { "start": "expo start", "reset-project": "node ./scripts/reset-project.js", "prebuild-android": "expo prebuild --clean --platform android", "android-device": "expo run:android --device", "build-android": "cd android && ./gradlew assembleRelease", "android": "expo run:android", "ios": "expo run:ios", "web": "expo start --web", "test": "jest --watchAll", "lint": "expo lint" }, "jest": { "preset": "jest-expo" }, "dependencies": { "@expo/react-native-action-sheet": "^4.1.1", "@expo/vector-icons": "^14.0.2", "@gorhom/bottom-sheet": "^5.1.4", "@hookform/resolvers": "^5.0.1", "@kolking/react-native-avatar": "^2.1.5", "@react-native-community/netinfo": "^11.4.1", "@react-navigation/bottom-tabs": "^7.3.13", "@react-navigation/native": "^7.1.9", "@rn-primitives/alert-dialog": "^1.1.0", "@rn-primitives/avatar": "^1.1.0", "@rn-primitives/dropdown-menu": "^1.1.0", "@rn-primitives/hooks": "^1.1.0", "@rn-primitives/label": "^1.1.0", "@rn-primitives/portal": "^1.2.0", "@rn-primitives/slot": "^1.1.0", "@rn-primitives/types": "^1.1.0", "@roninoss/icons": "^0.0.4", "@shopify/flash-list": "1.8.0", "@tanstack/react-query": "^5.76.1", "axios": "^1.9.0", "class-variance-authority": "^0.7.1", "clsx": "^2.1.1", "date-fns": "^4.1.0", "date-fns-tz": "^3.2.0", "eventemitter3": "^5.0.1", "expo": "~53.0.9", "expo-blur": "~14.1.4", "expo-camera": "~16.1.6", "expo-constants": "~17.1.6", "expo-dev-client": "~5.1.8", "expo-font": "~13.3.1", "expo-haptics": "~14.1.4", "expo-image": "~2.1.7", "expo-linking": "~7.1.5", "expo-navigation-bar": "~4.2.4", "expo-router": "~5.0.7", "expo-splash-screen": "~0.30.8", "expo-status-bar": "~2.2.3", "expo-symbols": "~0.4.4", "expo-system-ui": "~5.0.7", "expo-web-browser": "~14.1.6", "fast-xml-parser": "^5.2.3", "i18next": "^25.0.2", "lucide-react-native": "^0.503.0", "moti": "^0.30.0", "nativewind": "^4.1.23", "prettier-plugin-tailwindcss": "^0.6.11", "react": "19.0.0", "react-dom": "19.0.0", "react-hook-form": "^7.56.3", "react-i18next": "^15.2.0", "react-native": "0.79.2", "react-native-date-picker": "^5.0.12", "react-native-flash-message": "^0.4.2", "react-native-gesture-handler": "~2.25.0", "react-native-ios-context-menu": "^3.1.2", "react-native-ios-utilities": "^5.1.5", "react-native-keyboard-controller": "^1.17.1", "react-native-mmkv": "^3.2.0", "react-native-notifier": "^2.0.0", "react-native-page-indicator": "^2.4.2", "react-native-reanimated": "~3.17.5", "react-native-safe-area-context": "5.4.0", "react-native-screens": "~4.10.0", "react-native-svg": "15.12.0", "react-native-swiper-flatlist": "^3.2.5", "react-native-tab-view": "^4.1.0", "react-native-uitextview": "^2.0.3", "react-native-uuid": "^2.0.3", "react-native-web": "~0.20.0", "react-native-webview": "13.13.5", "scandit-datacapture-frameworks-core": "7.2.2", "scandit-react-native-datacapture-barcode": "7.2.2", "scandit-react-native-datacapture-core": "7.2.2", "scandit-react-native-datacapture-id": "7.2.2", "scandit-react-native-datacapture-parser": "7.2.2", "tailwind-merge": "^3.2.0", "tailwind-variants": "^1.0.0", "tailwindcss-animate": "^1.0.7", "xmlbuilder2": "^3.1.1", "zod": "^3.24.4", "zustand": "^5.0.4" }, "devDependencies": { "@babel/core": "^7.26.9", "@dev-plugins/react-query": "~0.3.1", "@types/jest": "^29.5.14", "@types/react": "~19.0.10", "eslint": "^9.25.0", "eslint-config-expo": "~9.2.0", "jest": "^29.7.0", "jest-expo": "~53.0.5", "prettier-plugin-tailwindcss": "^0.6.11", "tailwindcss": "^3.4.17", "typescript": "^5.8.3" }, "private": true } `

Your best off creating a full reproduceable repo instead of dumping a package.json.

**Comment by Titozzz** [comment]
https://github.com/Titozzz/header-button-bug

There is a repro

HOWEVER: this does not repro on all devices:
Broken on pixel 7 / Samsung S24
Working on Xiaomi Pocophone F1 / Oneplus Nord

**Comment by gorhom** [comment]
temporary workaround is using `TouchableOpacity`, `Pressable` form `react-native-gesture-handler` 🤷‍♂️

**Comment by pierroo** [comment]
I was hoping yesterday's release 2.26.0 would fix it, but not yet unfortunately :/

**Comment by jiroscripts** [comment]
The best workaround I found is to use onTouchEnd instead of onPress. I found this in an other Github issue. Hope this helps :)

**Comment by plastuchino** [comment]
I'm still having this issue. 

Here's the full repo so you can replicate the enviroment: [repo](https://github.com/XDagging/parking-app-front)

I'm having an issue where upon reload, the button perhaps works for 2-6 clicks before it stops working. This is the same for the textInput. I'm using expo with nativewind.

Doesn't work on s20 phone (android).


Here's what I've tried:
- Tried removing nativewind (nothing changed)
- Removed absolute positioning, modals, etc (doesn't work)
- Tried using onPress, onPressIn, onPressOut, onTouchEnd (didn't change issue)
- removed all fonts, used gestureHandlerRootView, removed SafeAreaView (nothing changed)
- created a new app from scratch with the base code with a pressable (still failed).

Honestly I'm just so lost and I've already spent so many hours on this. Would appreciate some help

**Comment by fkreinh** [comment]
@jiroscripts unfortunately it doesn't work on RN web

**Comment by fkreinh** [comment]
Hi @kkafar and @tboba 👋

I am kind of blocked on migrating to the new architecture because of this issue: on Android devices, Pressable’s onPress doesn’t work on native-stack specially with **TrueShet**

Would you be open to accepting a donation via GitHub Sponsors to help fund reopening and fixing this issue?

Thanks!

**Comment by joaqo** [comment]
Having the same issue, can't believe it's still not fixed.

**Comment by geroale** [comment]
Same issue here

**Comment by di-sukharev** [comment]
same issue

**Comment by alejandroleitz** [comment]
> The best workaround I found is to use onTouchEnd instead of onPress. I found this in an other Github issue. Hope this helps :)

This solve the issue thank you.

**Comment by Sascha195** [comment]
still an issue in [4.15.4](https://www.npmjs.com/package/react-native-screens/v/4.15.4)

**Comment by martinezguillaume** [comment]
I have some case when I can't use `onPress` on element outside of the header, it is very very rare but I can reproduce if i'm playing a bit with my app, pressing multiple buttons quickly.
in this blocking state, onTouchEnd and onPressIn work perfect.
So maybe it is a bit more complicated than juste the elements inside the header.
Only reproducible with Samsung physical device !

**Comment by kkafar** [comment]
I see this issue still lives on, despite the fact that we do not observe it on our end.

To work on the issue we need **reproduction**. Otherwise it is highly unlikely that something will be done here.

**Comment by maxencehenneron** [comment]
> I see this issue still lives on, despite the fact that we do not observe it on our end.
> 
> To work on the issue we need **reproduction**. Otherwise it is highly unlikely that something will be done here.

https://snack.expo.dev/@dernise/header-navigation-broken

Opening this snack on a real android device using expo go shows the problem when using headerTransparent: true. Does not happen when set to false.

**Comment by Abhishekk63** [comment]
> > The best workaround I found is to use onTouchEnd instead of onPress. I found this in an other Github issue. Hope this helps :)
> 
> This solve the issue thank you.

Yup! this works!

**Comment by jiroscripts** [comment]
For me, the issue is not only related to headers but also to pressable elements (such as Pressable and even TextInput) inside ScrollViews or form sheets. The issue occurs only on Android. Workarounds help a lot, but they all have drawbacks, and there are some scenarios with no acceptable solution.

**Comment by kkafar** [comment]
@jiroscripts please provide a reproduction, otherwise we won't move forward here.

**Comment by paour** [comment]
Wasn't [Maxence's repro](https://github.com/software-mansion/react-native-screens/issues/2219#issuecomment-3321197225) sufficient?

**Comment by kkafar** [comment]
We already patched it & landed a PR https://github.com/software-mansion/react-native-screens/pull/3239

**Comment by kkafar** [comment]
If you want to check it out, just install `react-native-screens@nightly`

**Comment by fkreinh** [comment]
@kkafar I can confirm this issue is still present with TrueSheet bottom sheets on Android devices. I've created a minimal reproduction repository that demonstrates the bug:

Repository: https://github.com/fkreinh/android-stack-pressable

Video Demo: https://www.loom.com/share/f8b7177fd2d7436c9097ae003923601b

APK for testing: https://expo.dev/accounts/felipe-krein/projects/android-stack-pressable/builds/50ed961d-959e-4bbd-a5cd-39e7fc1578a5

Setup:
React Native: 0.81.4
Expo: ~54.0.11
@lodev09/react-native-true-sheet: ^2.0.6
react-native-screens: nightly (4.17.0-nightly-20250930-6525eabd8)
Behavior:
✅ Pressable outside the TrueSheet bottom sheet works correctly
❌ Pressable inside the TrueSheet bottom sheet does not respond to touches on Android physical devices
✅ Both work fine on iOS and Android emulators
The issue appears to be related to how native view containers (like TrueSheet's native bottom sheet implementation) handle touch events on Android devices with the new architecture.

**Comment by jiroscripts** [comment]
@kkafar Is @fkreinh’s repro sufficient, or do you need more details to debug?

**Comment by kkafar** [comment]
@jiroscripts We do not maintain `react-native-true-sheet` here 😄 If you think our help is needed there you can contact us at [projects@swmansion.com](mailto:projects@swmansion.com).

Otherwise please report `react-native-true-sheet` issues to appropriate repository.

**Comment by jiroscripts** [comment]
@kkafar I actually have the same issue with my form sheets. I tried to make a repro from a fresh RN project, without success. I will try to update my project dependencies to check if there is any differences

**Comment by ChronoByteCosmonaut** [comment]
Still an issue on android apk :(

**Comment by balmasich** [comment]
@kkafar 
When open formSheet with sheetAllowedDetents: 'fitToContents',
Pressable element  with method onPress didn't working, but if you drag and try again press it works well
version "react-native-screens": "^4.19.0-nightly-20251119-cf5dc9305",

**Comment by hirvesh** [comment]
I am not sure if this helps, but the problem with onPress not working with scrollview seems tied to https://github.com/software-mansion/react-native-reanimated/issues/5977

I had the same issue which I thought was this issue here with react-native-screens, but was actually with re-animated. Upgrading to 4.x.x solved it (SDK 53).

I was looking in the wrong place (here for weeks).

**Comment by balmasich** [comment]
@hirvesh i don't have a reanimated

**Comment by balmasich** [comment]
"react-native": "0.81.5",
"expo": "~54.0.25",
"@react-navigation/native-stack": "7.6.3",
"react-native-safe-area-context": "^5.6.2",
android 15
new Arch enabled

**Comment by Nesh108** [comment]
Btw, it's still happening on 4.11.1 using "react-native-reanimated": "~3.17.4" (expo 53)

**Comment by jiroscripts** [comment]
I upgraded all dependencies to their latest versions with React Native 0.81.4, and we no longer report any issues.

**Comment by julientietkem** [comment]
The issue still appears
